File: app/src/main - 副本/java/com/zmcursor/mydaliy/Utils/TaskPool/Task.java
package com.zmcursor.mydaliy.Utils.TaskPool;


/**
 * Created by ZMcursor on 2018/6/1 0001.
 */

public abstract class Task implements Runnable, Comparable<Task> {

    private int prior;
    private long time;

    public Task() {
        this(5);
    }

    /**
     * @param prior int value between 1~10
     */
    public Task(int prior) {
        this.prior = prior;
        time = System.currentTimeMillis();
    }

    @Override
    public int compareTo(Task o) {
        return getValue(o) - getValue(this);
    }

    //    public void prepare() {
//        time = System.currentTimeMillis();
//    }

    //    public void setPrior(int prior) {
//        this.prior = prior;
//    }
    public int getPrior() {
        return prior;
    }

    public long getTime() {
        return time;
    }

    private int getValue(Task task) {
        return (int) ((task.getPrior() + (System.currentTimeMillis() - task.getTime() + 300)) / 1000);
    }
}

File: app/src/main/java/com/zmcursor/mydaliy/Utils/TaskPool/Task.kt
package com.zmcursor.mydaliy.Utils.TaskPool


/**
 * Created by ZMcursor on 2018/6/1 0001.
 */

abstract class Task
/**
 * @param prior int value between 1~10
 */
@JvmOverloads constructor(
        //    public void prepare() {
        //        time = System.currentTimeMillis();
        //    }

        //    public void setPrior(int prior) {
        //        this.prior = prior;
        //    }
        val prior: Int = 5) : Runnable, Comparable<Task> {
    val time: Long

    init {
        time = System.currentTimeMillis()
    }

    override fun compareTo(o: Task): Int {
        return getValue(o) - getValue(this)
    }

    private fun getValue(task: Task): Int {
        return ((task.prior + (System.currentTimeMillis() - task.time + 300)) / 1000).toInt()
    }
}
